.github/workflows/validate.yml

```diff
@@ -22,7 +22,7 @@ jobs:
 
       - name: Check for differences
         run: |
-          if ! git diff --quiet; then
+          if ! git diff --quiet -- . ':(exclude)stars.json'; then
```

Comment:
I don't think this should be required; if I'm reading it correctly, stars.json should not be modified by rebuild.sh